liquidity_pool/src/liquidity.rs

```diff
@@ -258,8 +269,12 @@ pub trait LiquidityModule:
 
         if asset_amount > &total_owed {
             let extra_asset_paid = asset_amount - &total_owed;
-            self.send()
-                .direct(&initial_caller, asset_token_id, 0, &extra_asset_paid, &[]);
+            self.send_fft_tokens(
+                &initial_caller,
+                &asset_token_id,
```

Comment:
`asset_token_id` is already a reference, so no need to borrow.
```suggestion
                asset_token_id,
```

Reply:
fixed